src/components/calendar/calendar.jsx

```diff
@@ -76,6 +76,24 @@ function Calendar(props) {
   const [activeDate, setActiveDate] = useState(defaultFocusedDate);
   let refs = {};
 
+  useEffect(() => {
+    if (selectedDate && defaultDate && !isSameDate(selectedDate, defaultDate)) {
+      setSelectedDate(defaultDate);
+      setFocusedDate(defaultDate);
+
+      setYear(highlightedDate.getFullYear());
+      setMonth(highlightedDate.getMonth());
+      setActiveDate(defaultDate);
+    }
+  }, [defaultDate]);
+
+  useEffect(() => {
+    if (activeDate.getMonth() !== month || activeDate.getFullYear() !== year) {
+      setMonth(activeDate.getMonth());
+      setYear(activeDate.getFullYear());
+    }
+  }, [activeDate]);
```

Comment:
Let's double check whether we need this useEffect.

- I think the Calendar was already working well without it (since this updates the highlighted date when the active date changes)
- I think updating the highlighted date state is already taken care of in the useEffect for `props.value` (above).

Reply:
This is needed when a user changes the date-input, we want the calendar's month / year to respond accordingly.